.github/workflows/dagster.yml

```
@@ -30,133 +56,81 @@ on:
         description: 'GCR service account credentials to push/pull Docker images'
         required: true
 
-env:
-  AUTH_URL: auth.stage-elm.overstory.ai
-
 jobs:
-  cleanup-pr:
-    if: github.event.action == 'closed' || ( github.event.action == 'unlabeled' && !contains(github.event.pull_request.labels.*.name, 'dev-env') )
-    runs-on: ubuntu-latest
-    steps:
-      - name: Set up Cloud SDK
-        uses: google-github-actions/setup-gcloud@v0.2.1
-        with:
-          project_id: ${{ inputs.gcp_project }}
-          service_account_key: ${{ secrets.GCR_RW_SERVICEACCOUNT_KEY }}
-          export_default_credentials: true
-      - uses: google-github-actions/get-gke-credentials@v0.3.0
-        with:
-          cluster_name: ${{ format('stage-{0}', inputs.cluster_name) }}
-          location: europe-west4
-      - name: 'Remove dev-environment'
-        uses: AndreaGiardini/helm@master # Should be replaced with deliverybot/helm@v1 when https://github.com/deliverybot/helm/issues/66 is fixed
-        with:
-          task: 'remove'
-          helm: helm3
-          repository: 'https://dagster-io.github.io/helm'
-          chart: 'dagster'
-          release: "${{ format('{0}-pr-{1}', inputs.image_name, github.event.number) }}"
-          namespace: 'dagster'
-      - uses: marocchino/sticky-pull-request-comment@v2
-        with:
-          header: dev-env-location
-          delete: true
 
-  docker:
-    if: github.event.action != 'closed'
+  run-docker-workflow:
+    # only run on synchronize PR event, meaning something has been pushed to the PR branch
+    if: github.event.action == 'synchronize'
+    # query our reusable docker build and push workflow
+    uses: 20treeAI/github-workflows/.github/workflows/docker_build_push.yml@remove-env-vars
+    with:
+      image_name: ${{ inputs.image_name }}
+      branch: ${{ inputs.branch }}
+      gcp_project: ${{ inputs.gcp_project }}
+      test_dagster: true
+    secrets:
+      SSH_KEY: ${{ secrets.SSH_KEY }}
+      GCR_RW_SERVICEACCOUNT_KEY: ${{ secrets.GCR_RW_SERVICEACCOUNT_KEY }}
+
+  deploy-to-gke:
+    if: github.event.action == 'synchronize'
```

Deploy to cluster should only run when we merge the PR (push to main) or release (push a new tag)
The `deploy-dev-env-to-gke` will take care of the deployment for PR syncs
```suggestion
    if: github.event_name == 'push'
```